src/python/pants/engine/legacy/graph.py

```diff
@@ -358,35 +364,33 @@ class HydratedTarget:
   of hashing: we implement eq/hash via direct usage of an Address field to speed that up.
   """
   address: Any
```

Comment:
I originally had this as `Address` but it resulted in a bunch of errors and looks like this ends up usually being a `BuildFileAddress`? The PR was already getting too big so I gave up on it.